docs/exports_yaml_files.md

```diff
@@ -0,0 +1,121 @@
+  plate_includes: wells.qc_results
+duplex_seq_pcr_xp_concentrations_for_custom_pooling:
+  csv: duplex_seq_pcr_xp_concentrations_for_custom_pooling
+  plate_includes: wells.qc_results
+  ancestor_purpose: LDS AL Lib Dil
+hamilton_aggregate_cherrypick:
+  csv: hamilton_aggregate_cherrypick
+  plate_includes: wells.transfer_requests_as_target.source_asset
+  workflow: Cherry Pick
+```
+
+The rest of the document describes the structure of this file, and what each of the keys do.
+
+## Top level
+
+Each file is a `.yml` file located in `app/config/exports`, it contains the
+configuration for one or more {Export exports}.
+
+The top level structure consists of series of keys, uniquely identifying each
+export. Keys need to be unique across _all_ exports, not just those within
+the same file. Limber will detect duplicate keys, and will raise an exception
+on boot.
+
+The values in turn are used to describe each {Export}. The valid options are details in Export below.
+
+### Export
+
+Each export is identified by a unique key, that forms the exports id. This key
+is used in a similar manner to a primary key of a database, and will let your
+export be found with `Export.find(id)`. This id will also form part of the url
+for the export, so should be something meaningful to an end user.
+
+This id should be referenced in the {file:docs/pipelines_yaml_files.md#file_links file links}
+section of a purpose config to generate the appropriate links.
+
+@see Export for the Ruby objects generated by this configuration.
+
+```yaml
+unique_export:
+  csv: unique_export
+  plate_includes: wells.transfer_requests_as_target.source_asset
+  workflow: Cherry Pick
+  ancestor_purpose: LDS AL Lib Dil
+```
+
+The other keys are detailed below.
+
+#### csv
+
+**[required]**
+
+Determines the template in `app/views/exports` that will be used to render the
+export. If possible, try and use the same value for the template and the id.
+
+```yaml
+csv: unique_export
+```
+
+This example will use the `unique_export.csv.erb` file in the
+`app/views/exports` directory.
+
+#### plate_includes
+
+plate_includes indicates a [JSON API](https://jsonapi.org) include request
+parameter to eager-load data while fetching a plate.
+
+[More details](https://jsonapi.org/format/#fetching-includes)
+
+```yaml
+plate_includes: wells.transfer_requests_as_target.source_asset
+```
+
+Default: `wells`
+
+#### workflow
+
+A simple string that can be presented in the rendered template to distinguish
+between otherwise identical processes.
+
+Examples in `hamilton_cherrypick_dilutions`
+
+```yaml
+workflow: Cherry Pick
+```
+
+#### ancestor_purpose
+
+If this is supplied, the {ExportsController} will identify the first ancestor
+(earliest) of the current plate with this purpose, and will assign it to
+the instance variable `@ancestor_plate`.
+
+Example in `app/views/exports/duplex_seq_pcr_xp_concentrations_for_custom_pooling.csv.erb`
+
+```yaml
+ancestor_purpose: LDS AL Lib Dil
+```
```

Comment:
Great docs!
Noticed we appear to have tests in specs/views/exports for each of the different 'types' of export file that we create (i.e. not one for every single csv export in the configs). Maybe worth documenting that if you create a new file type/format it needs a test there?

Reply:
That makes sense, as you're testing the view at that point. There's also some tests in the controller specs. I'll make a note pointing to these.